File: synchro-mes/backend/app/schemas/pmp.py
"""
Schemas: PMP (Moído/Borra/Sucata)
"""
from pydantic import BaseModel, Field
from datetime import datetime


class PmpEntryCreate(BaseModel):
    type: str  # moido, borra, sucata
    machine_code: str | None = None
    product_code: str | None = None
    material: str | None = None
    weight_kg: float = Field(gt=0)
    operator_name: str | None = None
    shift: str | None = None
    destination: str | None = None
    notes: str | None = None


class PmpEntryRead(BaseModel):
    id: int
    type: str
    machine_code: str | None = None
    product_code: str | None = None
    material: str | None = None
    weight_kg: float
    operator_name: str | None = None
    shift: str | None = None
    destination: str | None = None
    notes: str | None = None
    timestamp: datetime | None = None

    model_config = {"from_attributes": True}

File: synchro-mes/frontend/src/types/pmp.ts
// ── PMP (Moído / Borra / Sucata) ─────────────────────────────

export type PmpType = 'moido' | 'borra' | 'sucata';

export interface PmpEntry {
  id: number;
  type: PmpType;
  machine_code: string | null;
  product_code: string | null;
  material: string | null;
  weight_kg: number;
  operator_name: string | null;
  shift: string | null;
  destination: string | null;
  notes: string | null;
  timestamp: string | null;
}

export interface PmpEntryCreate {
  type: PmpType;
  machine_code?: string;
  product_code?: string;
  material?: string;
  weight_kg: number;
  operator_name?: string;
  shift?: string;
  destination?: string;
  notes?: string;
}
